app/models/concerns/taxonomix.rb

```diff
@@ -173,6 +173,8 @@ def children_of_selected_organization_ids
   # for existing records it would block saving global objects if taxonomies were not changed
   # for new records this would allow to create global objects for non global users
   def ensure_taxonomies_not_escalated
+    return true if self.respond_to?(:skip_taxonomy_escalation_check?) && self.skip_taxonomy_escalation_check?
```

Comment:
This is basically feature envy from this class to Filter. I think this can fixed more appropriately adding code only where it's concerned, in Filter:

```ruby
class Filter
  def ensure_taxonomies_not_escalated
    super if self.resource_class.included_modules.include?(Taxonomix)
  end
end
``` 

That seems to solve the issue locally w/o these checks that have to be done everywhere but matter to one class only